### PR TITLE
Refactor MTE-3341 Firefox iOS XCUITests for iOS 18.0

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3341)

## :bulb: Description
I would like to make the XCUITest to compatible with iOS 18 so that the team is prepared for the upcoming iOS release.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/NavigationTest.swift [hunk 1]
```diff
@@ -295,8 +295,10 @@ class NavigationTest: BaseTestCase {
         } else {
             mozWaitForElementToExist(app.buttons["Copy"])
         }
-        if !iPad() {
-            mozWaitForElementToExist(app.scrollViews.staticTexts["Messages"])
+        if #unavailable(iOS 18) {
+            if !iPad() {
+                mozWaitForElementToExist(app.scrollViews.staticTexts["Messages"])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/NavigationTest.swift:300]: There's no "Messages" option available in iOS 18.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ReadingListTests.swift [hunk 2]
```diff
@@ -235,7 +235,10 @@ class ReadingListTests: BaseTestCase {
         savedToReadingList.tap()
         // The article is displayed in Reader View
         mozWaitForElementToExist(app.buttons["Reader View"])
-        XCTAssertTrue(app.buttons["Reader View"].isSelected)
+        // iOS 18 only: Reader View icon is enabled but is not selected.
+        if #unavailable(iOS 18) {
+            XCTAssertTrue(app.buttons["Reader View"].isSelected)
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ReadingListTests.swift:240]: I'm not sure if this is a bug: The reader view icon from the awesome bar isn't selected anymore on iOS 18.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BrowsingPDFTests.swift [hunk 3]
```diff
@@ -115,7 +115,6 @@ class BrowsingPDFTests: BaseTestCase {
         navigator.openURL(PDF_website["url"]!)
         waitUntilPageLoad()
         navigator.performAction(Action.BookmarkThreeDots)
-        waitUntilPageLoad()
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/BrowsingPDFTests.swift:118]: The page isn't reloaded on iOS 18 for sure. Removing this line seems to work in older iOS versions.

**Review by dragosb01 — APPROVED** [review]

**Comment by clarmso** [comment]
@mergifyio backport release/v130

**Pull request merged**